src/mds/MDCache.cc

```diff
@@ -8515,7 +8515,7 @@ int MDCache::path_traverse(const MDRequestRef& mdr, MDSContextFactory& cf,
 	  lov.add_xlock(&dn->lock);
 	} else {
 	  // force client to flush async dir operation if necessary
-	  if (cur->filelock.is_cached())
+	  if (!mdr->lock_cache && cur->filelock.is_cached())
```

Comment:
Should we not simply check that the filelock is not already in the `lock_cache`?

Reply:
done.